src/librustc_mir/transform/const_prop.rs

```diff
@@ -669,28 +669,43 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
             let place_ty: Ty<'tcx> = place
                 .ty(&self.local_decls, self.tcx)
                 .ty;
-            if let Ok(place_layout) = self.tcx.layout_of(self.param_env.and(place_ty)) {
-                if let Some(value) = self.const_prop(rval, place_layout, statement.source_info) {
-                    if let Place {
-                        base: PlaceBase::Local(local),
-                        projection: box [],
-                    } = *place {
-                        trace!("checking whether {:?} can be stored to {:?}", value, local);
-                        if self.can_const_prop[local] {
-                            trace!("storing {:?} to {:?}", value, local);
-                            assert!(self.get_const(local).is_none());
-                            self.set_const(local, value);
-
-                            if self.should_const_prop() {
-                                self.replace_with_const(
-                                    rval,
-                                    value,
-                                    statement.source_info,
-                                );
+
+            match self.tcx.layout_of(self.param_env.and(place_ty)) {
+                Ok(place_layout) => {
+                    if let Some(value) = self.const_prop(
+                        rval,
+                        place_layout,
+                        statement.source_info,
+                    ) {
+                        if let Place {
+                            base: PlaceBase::Local(local),
+                            projection: box [],
+                        } = *place {
+                            trace!("checking whether {:?} can be stored to {:?}", value, local);
+                            if self.can_const_prop[local] {
+                                trace!("storing {:?} to {:?}", value, local);
+                                assert!(self.get_const(local).is_none());
+                                self.set_const(local, value);
+
+                                if self.should_const_prop() {
+                                    self.replace_with_const(
+                                        rval,
+                                        value,
+                                        statement.source_info,
+                                    );
                             }
                         }
                     }
                 }
+                Err(LayoutError::Unsized(_ty)) => {
+                    let sp = statement.source_info.span; // async fn block
+                    self.tcx.sess.struct_span_err(
```

Comment:
Uhm; `const_prop` is too my knowledge an optimization phase so why are we emitting an error here?

Reply:
Yea I think it's totally fine to just keep doing what the old code did and ignore errors. They may be occuring because we're in a generic function and may be bogus for specific concrete types

Reply:
Should we close this PR then?

Reply:
Well; we cannot merge this PR as-is I think but maybe we can morph the PR into something else. You can always open a new one tho... up to you. :)